.github/pull_request_template.md

```diff
@@ -1,10 +1,18 @@
-**Related Issue(s)**:  ...
+### Related Issues
+- fixes #issue-number
 
-**Proposed changes**:
-- ...
+### Proposed Changes?
+ <!--- In case of a bug: Describe what caused the issue and how you solved it-->
 
-## Pre-flight checklist
-- [ ]  I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md)
-- [ ] I have [enabled actions on my fork](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md#forks)
-- [ ] If this is a code change, I added tests or updated existing ones 
-- [ ] If this is a code change, I updated the docstrings
+### How did you test it?
+<!-- unit tests, integration tests, manual verification, instructions for manual tests -->
+
+### Notes for the reviewer
+<!-- E.g. point out section where the reviewer  -->
+
+### Checklist
+- [ ] I have updated the referenced issue with new insights and changes
+- [ ] If this is a code change, I have added unit tests
```

Comment:
Minor: why specifying "**unit**" tests? Some PRs deserve integration tests too, if not end2end in case of new nodes or substantial features. Sometimes contributors bring those too.

I believe we should remove "**unit**" from this line.

```suggestion
- [ ] If this is a code change, I have added tests
```